File: TransCoder_tokenized_test_set_functions/cpp/LARGEST_SUBSET_WHOSE_ALL_ELEMENTS_ARE_FIBONACCI_NUMBERS.cpp
void findFibSubset ( int arr [ ], int n ) {
  int max = * std :: max_element ( arr, arr + n );
  int a = 0, b = 1;
  unordered_set < int > hash;
  hash . insert ( a );
  hash . insert ( b );
  while ( b < max ) {
    int c = a + b;
    a = b;
    b = c;
    hash . insert ( b );
  }
  for ( int i = 0;
  i < n;
  i ++ ) if ( hash . find ( arr [ i ] ) != hash . end ( ) ) printf ( "%d ", arr [ i ] );
}

File: TransCoder_tokenized_test_set_functions/python/LARGEST_SUBSET_WHOSE_ALL_ELEMENTS_ARE_FIBONACCI_NUMBERS.py
def findFibSubset ( arr, n ) :
    m = max ( arr )
    a = 0
    b = 1
    hash = [ ]
    hash . append ( a )
    hash . append ( b )
    while ( b < m ) :
        c = a + b
        a = b
        b = c
        hash . append ( b )
    for i in range ( n ) :
        if arr [ i ] in hash :
            print ( arr [ i ], end = " " )
    
